src/lib/containers/DirectDownloadButton.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react'
+import { Button } from 'react-bootstrap'
+import { BatchFileRequest, FileHandleAssociation } from '../utils/synapseTypes'
+import { getFiles } from '../utils/SynapseClient'
+
+export type DirectFileDownloadButtonProps = {
+  id?: string
+  token: string
+  fileHandleAssociation: FileHandleAssociation
+  fileName: string | undefined
+  className?: string
+  variant?: string  // This allows you to change the look of the button (see react bootstrap doc)
+}
+
+const DirectDownloadButton: React.FC<DirectFileDownloadButtonProps> = props => {
+
+  const {id, token, fileHandleAssociation, className, variant, fileName}  = props
+  let mounted:boolean = true
+
+  useEffect( () => {
```

Comment:
Does this do anything?

Reply:
No except when the token is updated. You think it's okay to delete it?

Reply:
When the token is updated, what does this do?

Reply:
It will refresh the component. If the token is changed and it also refreshes the page, I guess I can delete this code.